tests/v2/validation/certrotation/cert_rotation.go

```diff
@@ -104,5 +150,243 @@ func rotateCerts(client *rancher.Client, clusterName string) error {
+		"client-" + clusterType + "-controller",
+
+		"client-scheduler",
+		// "client-supervisor", not in either list, need to check what is expected. not currently rotated
+		// "apiserver-loopback-client__.crt", need to check what is expected. on k3s
+		"etcd/client",
+		"etcd/peer-server-client",
+		"kube-controller-manager/kube-controller-manager",
+		"kube-scheduler/kube-scheduler",
+		// "server-ca.nochain.crt", not in either list, need to check what is expected. rotates, but not sure cadence (20 min after the others rotated)
+		"serving-kube-apiserver",
+
+		// ca certs aren't rotated
+		// caCertList := []string{"server-ca", "client-ca", "request-header-ca", "etcd/peer-ca", "etcd/server-ca"}
+	}
+
+	for _, filename := range certsList {
+		// ignoring errors since node roles have different subsets of the certs.
+		certString, _ := sshNode.ExecuteCommand("openssl x509 -enddate -noout -in " + certsPath + filename + certFileExtension)
+
+		if certString != "" {
+			certificates[filename] = certString
+		}
+	}
+
+	return certificates, nil
+}
+
+func getCertificatesFromV3Node(client *rancher.Client, v3Node *management.Node) (map[string]string, error) {
+	certificates := map[string]string{}
+
+	if !v3Node.ControlPlane {
+		return certificates, nil
+	}
+
+	sshNode, err := getSSHNodeFromV3Node(client, v3Node)
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Infof("Getting certificates from machine: %s", v3Node.ID)
+
+	certsPath := "/etc/kubernetes/ssl/"
+
+	certsList := []string{
+		"kube-apiserver",
+		"kube-apiserver-proxy-client",
+		// "kube-apiserver-requestheader-ca", ca isn't rotated
+		// "kube-ca", rotating but not renewing
+		//kubecfg not rotated
+		"kube-controller-manager",
+		"kube-node", // will not rotate on non-cp nodes
+		"kube-proxy",
+		"kube-scheduler",
+		// "kube-service-account-token", rotating, but not renewing
+		// certs/serverca not rotated
+	}
+
+	for _, filename := range certsList {
+		certString, _ := sshNode.ExecuteCommand("openssl x509 -enddate -noout -in " + certsPath + filename + pemFileExtension)
+
+		if certString != "" {
+			certificates[filename] = certString
+		}
+	}
+
+	return certificates, nil
+}
+
+func downloadRKE1SshKeys(client *rancher.Client, v3Node *management.Node) ([]byte, error) {
+	sshKeyLink := v3Node.Links["nodeConfig"]
+
+	req, err := http.NewRequest("GET", sshKeyLink, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Authorization", "Bearer "+client.RancherConfig.AdminToken)
+
+	resp, err := client.Management.APIBaseClient.Ops.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	privateSSHKeyRegEx := regexp.MustCompile(privateKeySSHKeyRegExPattern)
+	privateSSHKey := privateSSHKeyRegEx.FindString(string(bodyBytes))
+
+	return []byte(privateSSHKey), err
+}
+
+// GetSSHNodeFromMachine returns the v1/node object given a steve/v1/machine object.
```

Comment:
supernit - the function name is wrong in the comment.

Reply:
rm'd